sources/early-boot-config/user-data-providers/vmware-guestinfo/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+/// VMWare guestinfo
+#[macro_use]
+extern crate log;
+
+use early_boot_config_provider::provider::UserDataProvider;
+use early_boot_config_provider::{compression::OptionalCompressionReader, settings::SettingsJson};
+use serde::Deserialize;
+use snafu::{ensure, ResultExt};
+use std::io::{Cursor, Read};
+
+// The fields in which user data and its encoding are stored in guestinfo
+const GUESTINFO_USERDATA: &str = "guestinfo.userdata";
+const GUESTINFO_USERDATA_ENCODING: &str = "guestinfo.userdata.encoding";
+
+pub struct VmwareGuestinfo;
+
+#[cfg(target_arch = "x86_64")]
+impl VmwareGuestinfo {
```

Comment:
This still triggers [clippy failures](https://gist.github.com/bcressey/7997b8eb11d346cf0ea0e0e0faa62c56) on an aarch64 host.

Maybe try moving `lib.rs` to `x86_64.rs`, and having this become `lib.rs`:
```
 #[cfg(target_arch = "x86_64")]
mod x86_64;

#[cfg(not(target_arch = "x86_64"))]
mod not_x86_64;

#[cfg(target_arch = "x86_64")]
pub use x86_64::*;

#[cfg(not(target_arch = "x86_64"))]
pub use not_x86_64::*;
```

With `not_x86_64.rs`:
```
use early_boot_config_provider::settings::SettingsJson;
use early_boot_config_provider::provider::UserDataProvider;

pub struct VmwareGuestinfo;

impl UserDataProvider for VmwareGuestinfo {
    #[allow(dead_code)]
    fn user_data(&self) -> std::result::Result<Option<SettingsJson>, Box<dyn std::error::Error>> {
        unimplemented!()
    }
}
```

This seems to correctly no-op stuff on aarch64, but I haven't tested on x86_64. Then you can get rid of the `#[cfg(target_arch = "x86_64")]` conditionals in this file.